From 289b01a1b50ffe9bf4443f84f0c52fc2afce2c19 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 2 Jun 2017 17:39:45 +0800
Subject: [PATCH 1919/5242] MLK-15001-18 drm/imx: ldb: Revert an out-of-tree
 commit

commit  b9ba22e59361c52b2b41f24a38fd42c64862de1b from
https://source.codeaurora.org/external/imx/linux-imx.git

To follow the upstreaming implementation:

Revert commit de361832a52a ("MLK-12184 drm/imx: imx-ldb: add mux id check").

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/imx-ldb.c |   11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/imx/imx-ldb.c b/drivers/gpu/drm/imx/imx-ldb.c
index 3be4b47..3f02ee5 100644
--- a/drivers/gpu/drm/imx/imx-ldb.c
+++ b/drivers/gpu/drm/imx/imx-ldb.c
@@ -241,9 +241,8 @@ static void imx_ldb_encoder_enable(struct drm_encoder *encoder)
 		else if (imx_ldb_ch == &ldb->channel[1])
 			lvds_mux = &ldb->lvds_mux[1];
 
-		if (lvds_mux)
-			regmap_update_bits(ldb->regmap, lvds_mux->reg, lvds_mux->mask,
-					   mux << lvds_mux->shift);
+		regmap_update_bits(ldb->regmap, lvds_mux->reg, lvds_mux->mask,
+				   mux << lvds_mux->shift);
 	}
 
 	regmap_write(ldb->regmap, IOMUXC_GPR2, ldb->ldb_ctrl);
@@ -265,12 +264,6 @@ static void imx_ldb_encoder_enable(struct drm_encoder *encoder)
 	int mux = drm_of_encoder_active_port_id(imx_ldb_ch->child, encoder);
 	u32 bus_format = imx_ldb_ch->bus_format;
 
-	if (mux < 0) {
-		dev_warn(ldb->dev,
-			 "%s: cannot get valid mux id\n", __func__);
-		return;
-	}
-
 	if (mode->clock > 170000) {
 		dev_warn(ldb->dev,
 			 "%s: mode exceeds 170 MHz pixel clock\n", __func__);
-- 
1.7.9.5

